package.json

```diff
@@ -44,5 +44,7 @@
     "webpack": "^3.5.6",
     "xml-js": "^1.6.2"
   },
-  "dependencies": {}
+  "dependencies": {
+    "tslib": "^1.9.2"
```

Comment:
Can this be a devDependency?

Reply:
Not quite, the point of this PR is to output this:
```js
import * as tslib_1 from "tslib";

// ...

var stateNode = new StateNode(tslib_1.__assign({}, stateConfig, { key: key, parent: _this }));
```

instead of this:
```js
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};

// ...

var stateNode = new StateNode(__assign({}, stateConfig, { key: key, parent: _this }));
```

`tslib` is a real dependency that if used is supposed to be added to `"dependencies"` as it's imported/required

Reply:
@davidkpiano friendly 🏓 😉

Reply:
Hey, sorry for the delay. I want to avoid adding any dependencies to this package; is there any way this can be done without an extra dependency?

Reply:
> Hey, sorry for the delay. 

Not a problem 😃 

> I want to avoid adding any dependencies to this package; is there any way this can be done without an extra dependency?

There is not, because actual source code (and ofc compiled files) will refer to this dependency and we want people having it installed automatically to avoid missing deps problems.

Either way (with this PR merged or not) you still have a dependency on that code - it only may be explicit or implicit one, bundled in or kept in separate package.

Reply:
Would it be possible to add this at build time instead?  Perhaps xstate could be persuaded to build two artifacts, one with this dependency and one without?  Different npm package names, but from the same source?  That way, the code base is clean, but bundlers have an option that helps bundlers do their deduping.

Reply:
yes, it's doable - but IMHO it complicates things, users have to chose which flavour they want, for imho no good reason